st2common/st2common/hooks.py

```diff
@@ -166,8 +167,16 @@ class JSONErrorResponseHook(PecanHook):
 
     def on_error(self, state, e):
         error_msg = getattr(e, 'comment', str(e))
-        LOG.debug('API call failed: %s', error_msg)
-        LOG.debug(traceback.format_exc())
+        extra = {
+            'exception_class': e.__class__.__name__,
+            'exception_message': str(e),
+            'exception_data': e.__dict__
+        }
+
+        LOG.debug('API call failed: %s', error_msg, extra=extra)
+
+        if cfg.CONF.debug:
```

Comment:
Actually I was just surprised to see this flag being used. Not seen it in the code base anywhere else so this is a little bit surprising. 

Btw, logging exceptions at debug is perfectly fine - even if they are expected exceptions. Typical deployment scenarios would be logs at info level at least that is what we ship by default so should not expect anything else.